packages/dds/tree/src/test/tablePerformanceTestUtilities.ts

```diff
@@ -0,0 +1,186 @@
+ */
+const schemaFactory = new SchemaFactoryAlpha("test");
+
+/**
+ * Defines the schema for a table cell.
+ */
+export const Cell = schemaFactory.string;
+
+/**
+ * Defines the schema for a table column.
+ */
+export class Column extends TableSchema.column({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+/**
+ * Defines the schema for a table row.
+ */
+export class Row extends TableSchema.row({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+/**
+ * Defines the schema for a table, which includes columns and rows.
+ * It uses the previously defined Cell, Column, and Row schemas.
+ */
+export class Table extends TableSchema.table({
+	schemaFactory,
+	cell: Cell,
+	column: Column,
+	row: Row,
+}) {}
+
+/**
+ * Provides a simple table tree initialized with the specified size and cell value.
+ * This helper function creates a table schema, initializes a SharedTree instance,
+ * and populates it with the specified number of rows and columns.
+ * Each cell is initialized with the provided cell value.
+ *
+ * @param tableSize - The number of rows and columns to create in the table.
+ * @param cellValue - The initial value to set in each cell of the table.
+ * @returns A fully initialized table tree definition, including table instance, undo/redo stacks, and a cleanup function.
+ */
+export function createTableTree(tableSize: number, cellValue: string): TableTreeDefinition {
+	const undoStack: Revertible[] = [];
+	const redoStack: Revertible[] = [];
+	const sharedTreeFactory = DefaultTestSharedTreeKind.getFactory();
+	const runtime = new MockFluidDataStoreRuntime({
+		idCompressor: createIdCompressor(),
+		attachState: AttachState.Detached,
+	});
+	const tree = sharedTreeFactory.create(runtime, "tree");
+	const treeView = tree.viewWith(
+		new TreeViewConfiguration({
+			schema: Table,
+			enableSchemaValidation: true,
+		}),
+	);
+
+	treeView.initialize(Table.empty());
+
+	const table = treeView.root;
+	for (let i = 0; i < tableSize; i++) {
+		const column = new Column({});
+		table.insertColumn({ index: i, column });
+	}
+	for (let i = 0; i < tableSize; i++) {
+		const row = new Row({ cells: {} });
+		table.insertRow({ index: i, row });
+	}
+	for (const row of table.rows) {
+		for (const column of table.columns) {
+			table.setCell({
+				key: {
+					column,
+					row,
+				},
+				cell: cellValue,
+			});
+		}
+	}
+	return {
+		table,
+		undoStack,
+		redoStack,
+		treeView,
+	};
+}
+
+/**
+ * Currently table schema does not support removing cells when a column is removed.
+ * This function provides a way to remove a column and its associated cells from the table. Might remove in the future
+ * if the table schema is updated to handle this automatically.
+ */
+export function removeColumnAndCells(table: InstanceType<typeof Table>, column: Column): void {
+	Tree.runTransaction(table, () => {
+		table.removeColumn(column);
+		for (const row of table.rows) {
+			table.removeCell({ column, row });
+		}
+	});
+}
+
+export function populateUndoRedoStacks(
```

Comment:
I would recommend updating this to initialize and return the undo/redo stacks, rather than taking them in and mutating them.

Reply:
Removed this and replaced with undoRedoManager class